src/index.js

```diff
@@ -27,70 +27,63 @@ async function start(fields) {
 }
 
 async function getBillsListAndTestTokens(ovh) {
-  try {
-    return await ovh.requestPromised('GET', '/me/bill')
-  } catch (e) {
-    if (
-      e.error &&
-      e.error == 403 &&
-      e.message &&
-      (e.message == 'This application key is invalid' ||
-        e.message == 'This credential does not exist')
-    ) {
-      log('error', e)
-      throw new Error(errors.LOGIN_FAILED)
-    } else if (
-      e.error &&
-      e.error == 400 &&
-      e.message &&
-      e.message == 'Invalid signature'
-    ) {
-      log('error', e)
-      log('error', 'The AppSecret seems not valid')
-      throw new Error(errors.LOGIN_FAILED)
-    } else if (
-      e.error &&
-      e.error == 403 &&
-      e.message &&
-      e.message == 'This call has not been granted'
-    ) {
-      log('error', e)
-      log('error', 'Auth ok, but GET /me/bill have not been granted')
-      throw new Error(errors.LOGIN_FAILED)
-    } else if (
-      e.error &&
-      e.error == 403 &&
-      e.message &&
-      e.message == 'This credential is not valid'
-    ) {
-      log('error', e)
-      log('error', 'The tokens seems to have expired')
-      throw new Error(errors.LOGIN_FAILED)
-    } else {
+  return await ovh.requestPromised('GET', '/me/bill').catch(function(e) {
+    if (e.error == undefined || e.message == undefined) {
       throw e
-    }
-  }
+    } else {
+      if (
+        e.error == 403 && (
+          e.message == 'This application key is invalid' ||
+            e.message == 'This credential does not exist')) {
+        log('error', e)
+        throw new Error(errors.LOGIN_FAILED)
+      } else if (
+        e.error == 400 &&
+          e.message == 'Invalid signature'
+      ) {
+        log('error', e)
+        log('error', 'The AppSecret seems not valid')
+          throw new Error(errors.LOGIN_FAILED)
+      } else if (
+        e.error == 403 &&
+          e.message == 'This call has not been granted'
+      ) {
+        log('error', e)
+        log('error', 'Auth ok, but GET /me/bill have not been granted')
+        throw new Error(errors.LOGIN_FAILED)
+      } else if (
+          e.error == 403 &&
+          e.message == 'This credential is not valid'
+      ) {
+        log('error', e)
+        log('error', 'The tokens seems to have expired')
+        throw new Error(errors.LOGIN_FAILED)
+      } else {
+        throw e
+      }
+      }
+  })
```

Comment:
This is a very complete error management, but **it is hard to read**.
Maybe you can refactor in some way:
```js
// catch(e) { throw handleNetworkError(e); }

function handleNetworkError(e) {
  const message = e.message;
  const code = e.error;
  if (e.error === undefined || e.message === undefined) {
    return e;
  }
  log('error', e);
  if (message === "This call has not been granted") {
    log('error', 'Auth ok, but GET /me/bill have not been granted')
  }
  if (message === "This credential is not valid") {
    log('error', 'The tokens seems to have expired')
  }
  if (message === "Invalid signature") {
    log('error', 'The AppSecret seems not valid')
  }
  return new Error(errors.LOGIN_FAILED);
}
```

That should do almost (I did not write it on an editor and did not check if it is isofunctional) the same.